builder/dockerfile/builder.go

```diff
@@ -271,15 +271,16 @@ func (b *Builder) build(stdout io.Writer, stderr io.Writer, out io.Writer) (stri
 	}
 
 	// check if there are any leftover build-args that were passed but not
-	// consumed during build. Return an error, if there are any.
+	// consumed during build. Return a warning, if there are any.
 	leftoverArgs := []string{}
 	for arg := range b.options.BuildArgs {
 		if !b.isBuildArgAllowed(arg) {
 			leftoverArgs = append(leftoverArgs, arg)
 		}
 	}
+
 	if len(leftoverArgs) > 0 {
-		return "", fmt.Errorf("One or more build-args %v were not consumed, failing build.", leftoverArgs)
+		fmt.Fprintf(b.Stdout, "[Warning] One or more build-args %v were not consumed\n", leftoverArgs)
```

Comment:
A while ago someone posted a link to a web site that made an argument for when to use stdout vs stderr for things like info/warning messages. Their basic premise was that stdout should be reserved for normal/expected output - and therefore including things like info/warning messages in there would do nothing but muck-up any output processing (like grep, sed, ...) that people may have. It convinced me that stderr should be used.  But YMMV